CRM/Core/Smarty/plugins/block.crmPermission.php

```diff
@@ -32,7 +32,15 @@
  */
 function smarty_block_crmPermission($params, $content, &$smarty, &$repeat) {
   if (!$repeat) {
-    if (CRM_Core_Permission::check($params['permission'], $params['contact_id'] ?? NULL)) {
+    if (empty($params['permission']) && empty($params['not'])) {
+      // This would be due to developer error - better to return nothing to make it more visible.
+      return '';
+    }
+    $hasPermission = empty($params['permission']) || CRM_Core_Permission::check($params['permission'], $params['contact_id'] ?? NULL);
+    if (!$hasPermission) {
+      return '';
+    }
+    if (empty($params['not']) || !CRM_Core_Permission::check($params['permission'], $params['contact_id'] ?? NULL)) {
```

Comment:
@eileenmcnaughton now that I look closely, the logic here looks wrong. The second condition is going to short-circuit the third.
Also, semantically I think `has` & `not` would be nice substitutes for `permission`. So you could do:
```
{crmPermission has="access CiviCRM" not="administer CiviCRM"}
  This block applies to regular users who are NOT administrators!
{/crmPermission}
```